internal/services/sql/sql_managed_instance_failover_group_resource.go

```diff
@@ -0,0 +1,360 @@
+			return fmt.Errorf("setting `readonly_endpoint_failover_policy_enabled`: %+v", err)
+		}
+
+		d.Set("role", string(props.ReplicationRole))
+
+		partnerManagedInstanceId, err := flattenSqlInstanceFailoverGroupManagedInstance(props.ManagedInstancePairs)
+		if err != nil {
+			return fmt.Errorf("flatten `partner_managed_instance_id`: %+v", err)
+		}
+		if err := d.Set("partner_managed_instance_id", partnerManagedInstanceId); err != nil {
+			return fmt.Errorf("setting `partner_managed_instance_id`: %+v", err)
+		}
+
+		if err := d.Set("partner_region", flattenSqlInstanceFailoverGroupPartnerRegions(props.PartnerRegions)); err != nil {
+			return fmt.Errorf("setting `partner_region`: %+v", err)
+		}
+	}
+
+	return nil
+}
+
+func resourceSqlInstanceFailoverGroupDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Sql.InstanceFailoverGroupsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.InstanceFailoverGroupID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	future, err := client.Delete(ctx, id.ResourceGroup, id.LocationName, id.Name)
+	if err != nil {
+		return fmt.Errorf("deleting %q: %+v", id, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for deletion of %q: %+v", id, err)
+	}
+
+	return err
+}
+
+func expandSqlInstanceFailoverGroupReadWritePolicy(d *pluginsdk.ResourceData) *sql.InstanceFailoverGroupReadWriteEndpoint {
+	vs := d.Get("read_write_endpoint_failover_policy").([]interface{})
+	v := vs[0].(map[string]interface{})
+
+	mode := sql.ReadWriteEndpointFailoverPolicy(v["mode"].(string))
+	graceMins := int32(v["grace_minutes"].(int))
+
+	policy := &sql.InstanceFailoverGroupReadWriteEndpoint{
+		FailoverPolicy: mode,
+	}
+
+	if mode != sql.ReadWriteEndpointFailoverPolicyManual {
+		policy.FailoverWithDataLossGracePeriodMinutes = utils.Int32(graceMins)
+	}
+
+	return policy
+}
+
+func flattenSqlInstanceFailoverGroupReadWritePolicy(input *sql.InstanceFailoverGroupReadWriteEndpoint) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	policy := make(map[string]interface{})
+
+	policy["mode"] = string(input.FailoverPolicy)
+
+	if input.FailoverWithDataLossGracePeriodMinutes != nil {
+		policy["grace_minutes"] = *input.FailoverWithDataLossGracePeriodMinutes
+	}
+	return []interface{}{policy}
+}
+
+func expandSqlInstanceFailoverGroupReadOnlyPolicy(d *pluginsdk.ResourceData) *sql.InstanceFailoverGroupReadOnlyEndpoint {
+	mode := sql.ReadOnlyEndpointFailoverPolicyDisabled
+	if d.Get("readonly_endpoint_failover_policy_enabled").(bool) {
+		mode = sql.ReadOnlyEndpointFailoverPolicyEnabled
+	}
+	return &sql.InstanceFailoverGroupReadOnlyEndpoint{
+		FailoverPolicy: mode,
+	}
+}
+func expandSqlInstanceFailoverGroupManagedInstancePairs(d *pluginsdk.ResourceData, primaryID parse.ManagedInstanceId) *[]sql.ManagedInstancePairInfo {
```

Comment:
This either needs to be removed or reworked to use `partner_managed_instance_id`